lib/index.ts

```diff
@@ -536,11 +536,34 @@ export class Worker {
 				);
 
 				// TS-TODO: Fix these "any" castings
-				const result = formulaParser.evaluateObject(
+				const evaluatedContract = formulaParser.evaluateObject(
 					typeContract.data.schema,
 					objectWithLinks as any,
 				);
-				return kernel.insertContract(logContext, insertSession, result as any);
+				const result = await kernel.insertContract(
+					logContext,
+					insertSession,
+					evaluatedContract as any,
+				);
+				// This is the first time the contract has been inserted
+				// Create a link between the inserted contract and the creator of the contract
+				// Create link and insert
+				await kernel.insertContract(logContext, insertSession, {
```

Comment:
This might should be done in autumndb if we want to create a link for *all* contracts on insert. Also we should be careful about how we define the link relationship. We would normally have a definition containing both contract types and the name and inverse name of the relationship. If we do this for all contracts, I'm not sure what the target contract type definition would look like. Also the from won't always necessarily be a user, could be a "robot" of some type as well. I'd catch up with @ramirogm as he's working on relationships at the moment.

Reply:
@joshbwlng I'm not convinced that this should be done in autumndb. I agree that `from` can be any entity that can operate as ana ctor (which include "loops" right now). @inwenc Can you bring this topic to an architecture call? I think there are some good questions here that we need to resolve.

Reply:
@LucianBuzzo @joshbwlng  Noted for architecture call.

Reply:
@LucianBuzzo I'm not necessarily pushing that this should be done in autumndb, if anything I'm trying to understand the reasoning as to why it should be in the worker if all necessary components are available at the `autumndb` level. Perhaps as a product we don't want to push all consumers of `autumndb` to follow this specific pattern?

Reply:
#summary After discussion in the arch call, final decision is to add the links to actor code in JF-worker and keep AutumnDB out of JF concerns.